apps/web/src/components/views/SettingsView/components/SettingsLanding.tsx

```diff
@@ -1,50 +1,181 @@
+"use client"
 import AvatarImgChange from "@/app/settings/components/ChangeAvatar"
-import { CountryCodeSelect } from "@/app/settings/components/CountryCodeSelect"
-import { TextInput, Textarea } from "@dotkomonline/ui"
+import { trpc } from "@/utils/trpc/client"
+import type { UserWrite } from "@dotkomonline/types"
+import { Button, PhoneInput, TextInput, Textarea, isValidPhoneNumber } from "@dotkomonline/ui"
+import { zodResolver } from "@hookform/resolvers/zod"
 import type { NextPage } from "next"
 import type { User } from "next-auth"
+import { useState } from "react"
+import { useForm } from "react-hook-form"
+import { z } from "zod"
 
-interface FormInputProps {
-  title: string
-  children?: JSX.Element
-}
+const nameSchema = z
+  .string()
+  .min(2, "Navn må være minst 2 bokstaver")
+  .max(50, "Navn kan maks være 50 bokstaver")
+  .regex(/^[a-zA-ZæøåÆØÅ\s'-]+$/, "Navn kan ikke inneholde tall eller spesialtegn bortsett fra - og '")
+  .toLowerCase()
+  .trim()
+  .transform((v) =>
+    v
+      .split(" ")
+      .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
+      .join(" ")
+  )
 
-const FormInput: React.FC<FormInputProps> = ({ title, children }) => (
-  <div className="w-full border-t-[1px] border-slate-7 flex py-8 justify-between px-4">
-    <div className="w-1/4">{title}:</div>
-    <div className="flex-1 flex justify-center">{children}</div>
-  </div>
-)
+const updateUserSchema = z.object({
+  givenName: nameSchema,
+  familyName: nameSchema,
+  email: z.string().email("Ikke en gyldig epost"),
+  phone: z
+    .string()
+    .refine((v) => isValidPhoneNumber(v), "Ikke et gyldig telefonnummer")
+    .optional(),
+  allergies: z.array(z.string()).optional(),
+  picture: z.string().optional(),
+})
 
 const Landing: NextPage<{ user: User }> = ({ user }) => {
+  const [allergies, setAllergies] = useState<string>()
+  const updateUser = trpc.user.update.useMutation()
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    trigger,
+    formState: { errors, dirtyFields, isSubmitting },
+  } = useForm<UserWrite>({
+    resolver: zodResolver(updateUserSchema),
+    defaultValues: {
+      ...user,
+    },
+  })
+
+  const formatAllergies = (allergies: string) => {
+    setAllergies(
+      allergies
+        .toLowerCase()
+        .replace(/[\s,]+/g, ",")
+        .replace(/(^\w|,\w)/g, (match) => match.toUpperCase())
```

Comment:
Allergies has to be consistently either a single string-field or a list as presented to users, I don't think we should split it based on commas/spaces, users will just give input that does not turn well into a list

Reply:
1. I'm not sure if i understand the problem? If the allergies are used as a string, then most likely all users allergies will be unique and there will be no way of using them for analytics, for example knowing how many attendees in an event are allergic to nuts.
However, it might be a suitable solution to give the user an input-field for ONE allergy, then hitting a `+` to get another input-field for the next allergy.
3. What do you mean by **_users will just give input that does not turn well into a list_**, could you give an example.? In theory, a user can always give bad inputs even if the input is a string or a list

Reply:
> users allergies will be unique and there will be no way of using them for analytics, for example knowing how many attendees in an event are allergic to nuts.

You would need to perform some kind of text-search w/ synonyms. There is native support for that in e.g. postgres. Also, we don't do such kinds of analytics / statistics, this information is generally only useful on an individual level.

> However, it might be a suitable solution to give the user an input-field for ONE allergy, then hitting a + to get another input-field for the next allergy.

Yes exactly, but from the images in the PR-description it is shown as a text-field, but stored as a list

> could you give an example.?

Look at e.g. the values people have for allergies at old events like https://old.online.ntnu.no/dashboard/events/2268/, there is mix of description and extra contextual information, not just "nuts" / "peanuts" / "pork", and full sentences with weird splits.

IMO: trying to make a regex/split this is any way is a lost cause, just let it be a free text-field, and allow event organizers to sort by non-empty values just read through all of them individually. That is how it currently works, and I think most other features for this is over-engineering or complicating it without giving us any gains. A _potential_ alternative would be that we have a _specific_ list of allergies we _can_ accomodate for, and not allow user-input as text, but IMO that sounds like a lot of work with

- gathering the specific allergies
- makes the user have to actively hunt for their allergies
- maintaining/updating the list